game/src/ui/gameplay/Arena/tile-selector/TileSelector.gd

```diff
@@ -79,11 +79,23 @@ func take_action():
 	var tower_inventory = arena.get_node("UI/Control/TowerInventory")
 	match _action:
 		ACTION.BUILDING:
-			arena.build_tower(_x, _y, _type)
 			var mouse_tower = self.get_node("Tower")
+			var tower_cost = mouse_tower.gold_cost
+			var negative_tower_cost = tower_cost * -1
```

Comment:
Is there a reason u want to do this here?

I think its bad UI to have the unaffordable towers clickable, but not buildable.

Reply:
https://github.com/quantum-defence/quantum-defence/issues/90